File: Data Structures & Algorithms/two-integer-sum-ii/submission-1.py
class Solution:
    def twoSum(self, numbers: List[int], target: int) -> List[int]:
        left = 0
        right = len(numbers)-1

        while left < right:
            z = numbers[left] + numbers[right]
            if z == target:
                return [left+1, right+1]
            elif z > target:
                right -= 1
            else:
                left += 1
        return []

File: Data Structures & Algorithms/minimum-size-subarray-sum/submission-1.java
class Solution {
    public int minSubArrayLen(int target, int[] nums) {
        int minWindow = Integer.MAX_VALUE;
        int left = 0;
        int currentSum = 0;
        for(int right = 0;  right < nums.length; ++right){
            currentSum += nums[right];
            while(currentSum >= target && left <= right){
                minWindow = Math.min(minWindow, right - left + 1);
                currentSum -= nums[left];
                ++left;
            }
        }
        if(minWindow == Integer.MAX_VALUE){
            minWindow = 0;
        }
        return minWindow;

    }
}